public/reorderSiteAwards.php

```diff
@@ -10,46 +10,171 @@
+        });
+}
+
+function handleRowDragStart(event) {
+    currentGrabbedRow = event.target;
+    event.target.style.opacity = '0.3';
+}
+
+function handleRowDragEnd(event) {
+    currentGrabbedRow = null;
+    event.target.style.opacity = '1';
+}
+
+function handleRowDragEnter(event) {
+    const targetRow = event.target.closest('tr');
+    if (targetRow && currentGrabbedRow.parentNode === targetRow.parentNode) {
+        targetRow.classList.add('border');
+        targetRow.classList.add('border-menu-link');
+    }
+}
+
+function handleRowDragOver(event) {
+    event.preventDefault();
+    return false;
+}
+
+function handleRowDragLeave(event) {
+    const targetRow = event.target.closest('tr');
+    if (targetRow) {
+        targetRow.classList.remove('border');
+        targetRow.classList.remove('border-menu-link');
+    }
+}
+
+function handleRowDrop(event) {
+    event.preventDefault();
+
+    const dropTarget = event.target.closest('tr');
+
+    if (currentGrabbedRow && dropTarget) {
+        const draggedTable = currentGrabbedRow.closest('table');
+        const dropTargetTable = dropTarget.closest('table');
+
+        if (draggedTable === dropTargetTable) {
+            const dropTableId = event.target.closest('table').id;
+            const initialDisplayOrderValue = getInitialDisplayOrderValue(dropTableId);
+
+            const draggedRowIndex = Array.from(currentGrabbedRow.parentNode.children).indexOf(currentGrabbedRow);
+            const dropTargetIndex = Array.from(dropTarget.parentNode.children).indexOf(dropTarget);
+
+            if (draggedRowIndex !== dropTargetIndex) {
+                if (draggedRowIndex < dropTargetIndex) {
+                    dropTarget.parentNode.insertBefore(currentGrabbedRow, dropTarget.nextSibling);
+                } else {
+                    dropTarget.parentNode.insertBefore(currentGrabbedRow, dropTarget);
+                }
+
+                updateDisplayOrderValues(dropTableId, initialDisplayOrderValue);
+            }
+        }
+    }
+
+    dropTarget.classList.remove('border');
+    dropTarget.classList.remove('border-menu-link');
+}
+
+function getInitialDisplayOrderValue(dropTableId) {
+    let initialDisplayOrderValue = 0;
+
+    const tableRows = document.querySelectorAll(`#${dropTableId} tbody tr`);
+    if (tableRows[0]) {
+        const firstRowOrderInput = tableRows[0].querySelector('.displayorderedit');
+        initialDisplayOrderValue = parseInt(firstRowOrderInput.value, 10);
+    }
+
+    return initialDisplayOrderValue;
+}
+
+function updateDisplayOrderValues(dropTableId, startingValue = 0) {
+    const tableRows = document.querySelectorAll(`#${dropTableId} tbody tr`);
+
+    tableRows.forEach((row,index) => {
+        const displayOrderInput = row.querySelector('.displayorderedit');
+        const currentValue = parseInt(displayOrderInput.value, 10);
+
+        if (currentValue !== -1) {
+            displayOrderInput.value = startingValue + (10 * index);
```

Comment:
I'm not sure how much I like this solution. If I'm just swapping two badges, it seems unnecessary to reassign values for all the rest of them.

It also overwrites any manual changes I've made but not committed. i.e. given:
```
A   1
B   2
C   10
D   15
E   20
```
If I explicitly set D to 9, then drag A below B, this is what I end up with:
```
B   1
A   11
C   21
D   31
E   41
```
The fact that I wanted D before C is lost (and should maybe have been updated when I manually modified it).

It seems like having drag/drop + editable fields is undesirable. Maybe the editable field should be hidden and just reflect the indices? But then we have to decide how to handle hiding badges (negative indices) and group ordering (groups are ordered by the lowest visible badge index between groups).

Regarding hidden items: If I set my first item to be -99 (or and item to -99 and refresh), then any dragging operations automatically make the second item -89 and the third -79, etc., causing 9 additional items to be hidden.

Reply:
Where I ultimately landed is "Display Order" is an implementation detail and is now hidden from the user.